### PR TITLE
test: optimize session&planner/core unit test 

### PR DESCRIPTION
**Pull request by glorv**
What problem does this PR solve?
make several package unit test support run unit test parallel to speed up unit test running

What is changed and how it works?
optimize session&planner/core unit test and fix parallel run test problem.
- split session `testSessionSuite` into 2 test suites
- enable session & planner/core  unit test run in parallel by move cases depend on changing global variables into  serialSuite 
- fix several case that change global variables directly while it should be wrap by lock

optimize result:
```
>  go test -v -vet=off -p 10 -race github.com/pingcap/tidb/planner/core
ok  	github.com/pingcap/tidb/planner/core	116.440s

> go test -v -vet=off -p 10 -race github.com/pingcap/tidb/session
ok  	github.com/pingcap/tidb/session	93.397s
```

After change:
```
> go test -v -vet=off -p 10 -race github.com/pingcap/tidb/planner/core -check.p=true
ok  	github.com/pingcap/tidb/planner/core	31.602s

> go test -v -vet=off -p 10 -race github.com/pingcap/tidb/session -check.p=true
ok  	github.com/pingcap/tidb/session	65.009s
```

Check List
Tests

unit tests
Code changes

Side effects

Related changes

### CODE REVIEW

**Comment by glorv** [comment]
@sykp241095  PTAL

File: session/tidb.go [hunk 1]
```diff
@@ -118,6 +118,19 @@ var (
 	statsLease = int64(3 * time.Second)
 )
 
+func setStoreBootstrap(storeUUID string) {
```
Comment (zimulala) [inline session/tidb.go]: These codes are best not to be placed here. Maybe we can put it to util/testutil that dedicates to testing.

Comment (glorv) [reply session/tidb.go]: function `finishBootstrap` in also call this to set the flag

File: session/tidb.go [hunk 2]
```diff
@@ -118,6 +118,19 @@ var (
...
+	storeBootstrappedLock.Lock()
+	defer storeBootstrappedLock.Unlock()
+	storeBootstrapped[storeUUID] = true
+}
+
+// used for test
```
Comment (zimulala) [inline session/tidb.go]:
```suggestion
// unsetStoreBootstrap  is only used for tests.
```

**Comment by glorv** [comment]
@zz-jason  PTAL

**Pull request merged**